docs/variants.rst

```diff
@@ -86,26 +101,138 @@ The option is itself a map with the following keys:
 ``short``
     A short human-readable string to describe the option.
 
-``long``
-    A lengthier human-readable string to describe the option. This value is
-    optional.
+``long`` (Optional)
+    A lengthier human-readable string to describe the option.
 
-``buildType``
+``buildType`` (Optional)
```

Comment:
Is it possible to use `description`, `short`, `long`, `buildtype`, `linkage`, `settings `and `env ` in the setting part, outside of the choices. If not then please describe setting and the allowed choice paramter separate.

A example is shown below:

   ```
 buildType:
      default: debug,
      linkage: shared  #<----
      choices:
        debug:
          short: Debug
          long: Emit debug information
          buildType: Debug
        release:
          short: Release
          long: Optimize generated code
          buildType: Release
        .....
```